CMakeLists.txt

```diff
@@ -21,22 +27,79 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
+# =============================================================================
+# Option defaults.
+# =============================================================================
+# This allows h.nrnversion(6) to print only the configuration differences
+# analogously to what happens with an autotools build.
+set(NRN_ENABLE_SHARED_default                   ON)
```

Comment:
Can we use `_DEFAULT` suffix instead of `_default` everywhere? Just that mix of case doesn't look good (to me).

Reply:
I wondered about that. I was trying to indicate that the variables were entirely internal and not
to be changed from the command line but only in CMakeLists.txt itself. I'll raise to uppercase and add a more elaborate comment in CMakeLists.txt

Reply:
yeah, maybe let's not mix case. the fact that there is `_DEFAULT` in the name should make it clear that these are internal default values.

You can maybe move them into a seperate `Defaults.cmake` file into the cmake subdir?